### PR TITLE
implemented UserDetails and UserDetailsServiceImpl, and control of re…

### PR DESCRIPTION
**Pull request by Daniel1004K**
…quests

### CODE REVIEW

**Review by vKutik — APPROVED** [review]
Good job!

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .and()
+                .authorizeRequests()
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:33]:
```suggestion
                .authorizeRequests()
```

File: src/main/java/mate/academy/spring/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public void setName(RoleName name) {
+        this.name = name;
+    }
+
+    @Override
+    public String getAuthority() {
+        return null;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/model/Role.java:48]: let's remove this method

File: src/main/java/mate/academy/spring/model/User.java [hunk 3]
```diff
@@ -14,6 +19,19 @@
     @Column(unique = true)
     private String email;
     private String password;
+    @ManyToMany(fetch = FetchType.LAZY)
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/model/User.java:22]:
```suggestion
    @ManyToMany
```

LAZY is the default value

File: src/main/java/mate/academy/spring/service/impl/UserDetailsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserService userService;
+
+    public UserDetailsServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/spring/service/impl/UserDetailsServiceImpl.java:22]: can we use `orElseThrow(() -> ...)` here?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**